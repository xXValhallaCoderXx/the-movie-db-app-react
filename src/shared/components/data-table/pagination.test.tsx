import React from "react";
import {shallow} from "enzyme";
import Pagination from "./pagination";

describe("Data Table Component", () => {
  const pages = [{id: "1"}, {id: "2"}];

  const onClick = jest.fn();
  it("should render correctly", () => {
    const component = shallow(<Pagination pages={pages} goTo={onClick} />);
    expect(component).toMatchSnapshot();
  });
  it("should render correct number of buttons", () => {
    const component = shallow(<Pagination pages={pages} goTo={onClick} />);

    const pagination = component.find("#pagination");
    expect(pagination).toHaveLength(1);

    const buttons = component.find("button");
    expect(buttons).toHaveLength(pages.length);
  });
  it("should render correct content in buttons", () => {
    const component = shallow(<Pagination pages={pages} goTo={onClick} />);

    const buttons = component.find("button");
    expect(buttons).toHaveLength(pages.length);

    buttons.forEach((btn, idx) => {
      expect(btn.text()).toEqual((idx + 1).toString());
    });
  });
});
