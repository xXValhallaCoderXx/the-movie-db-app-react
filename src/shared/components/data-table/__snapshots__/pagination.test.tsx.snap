// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Component should render correctly 1`] = `
<div
  id="pagination"
>
  <button
    className="bg-gray-400 hover:bg-gray-700 text-black font-bold py-2 px-4 m-1"
    key="0"
    onClick={[MockFunction]}
    value={0}
  >
    1
  </button>
  <button
    className="bg-gray-400 hover:bg-gray-700 text-black font-bold py-2 px-4 m-1"
    key="1"
    onClick={[MockFunction]}
    value={1}
  >
    2
  </button>
</div>
`;
